store/proof.go

```diff
@@ -0,0 +1,178 @@
+package store
```

Comment:
could we spin the store pkg changes into its own pr, will help reduce the size of this pr

Reply:
Yes, of course.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/18633